dump_test.go

```diff
@@ -130,3 +130,89 @@ func TestDumpSeeker(t *testing.T) {
 		})
 	}
 }
+
+func TestUnzipNpz(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		wantFileName []string
+		wantShape    [][2]int
+		wantData     [][]float64
+	}{
+		{
+			name:         "testdata/data_float64_corder.npz",
+			wantFileName: []string{"arr1.npy", "arr0.npy"},
+			wantShape: [][2]int{
+				{6, 1}, {2, 3},
+			},
+			wantData: [][]float64{
+				{0, 1, 2, 3, 4, 5}, {0, 1, 2, 3, 4, 5},
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			f, err := os.Open(tc.name)
+			if err != nil {
+				t.Fatalf("could not open %q: %+v", tc.name, err)
+			}
+			defer f.Close()
+			files, err := UnzipNpz(f)
+			if err != nil {
+				t.Fatalf("could not unzip %q: %+v", tc.name, err)
+			}
+			for i, file := range files {
+				if got, want := file.Name, tc.wantFileName[i]; got != want {
+					t.Fatalf(
+						"filename mismatch:\ngot:\n%s\nwant:\n%s\n",
+						got, want,
+					)
+				}
+				r, err := file.Open()
+				if err != nil {
+					t.Fatalf("could not open %q: %+v", file.Name, err)
+				}
+				defer r.Close()
+				npyReader, err := NewReader(r)
+				if err != nil {
+					t.Fatalf("could not read %v: %+v", file.Name, err)
+				}
+				if got, want := npyReader.Header.Descr.Shape, tc.wantShape[i]; len(got) != 2 || got[0] != want[0] || got[1] != want[1] {
+					t.Fatalf(
+						"shape  mismatch in file %s :\ngot:\n%v\nwant:\n%v\n", file.Name,
+						got, want,
+					)
+				}
+				length := npyReader.Header.Descr.Shape[0] * npyReader.Header.Descr.Shape[1]
+				var raw = make([]float64, length)
+				err = npyReader.Read(&raw)
+				if err != nil {
+					t.Fatalf("could not read %q: %+v", file.Name, err)
+				}
+				if got, want := raw, tc.wantData[i]; !equal(got, want) {
```

Comment:
this could probably be written using `reflect.DeepEqual` instead of this ad-hoc `equal` function, right?